File: find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        
        if(nums.empty()) return {};
        int abs_diff;
        vector<int> result;
        
        for(int i=0;i<nums.size();i++){
            
            abs_diff=abs(nums[i])-1;
            
            if(nums[abs_diff] > 0){
                nums[abs_diff] *= -1;
                
            }
             
        }
        
        for(int i=0;i<nums.size();i++){
            
            if(nums[i]>0){
                result.push_back(i+1);
            }
            
        }
        return result;
        
        
    }
};

File: find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        
        answer=[]
        
        for i in range(len(nums)):
            a=abs(nums[i])-1
            if nums[a] > 0:
                nums[a]=nums[a]*-1
            # print(nums[i])
            
        for i in range(len(nums)):
            if(nums[i]>0):
                answer.append(i+1)
                # print(i)
                
        return answer
            
        
        
        